src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs

```diff
@@ -175,22 +171,39 @@ private string GetRepoUri()
         ///     quite expressive enough for the options
         /// </summary>
         /// <returns>True if they are being used properly, false otherwise.</returns>
-        private bool ValidateRootBuildOptions()
+        private bool ValidateRootBuildsOptions()
         {
+            bool hasNonIdRootBuildOptions =
+                (!string.IsNullOrEmpty(_options.RepoUri) ||
+                 !string.IsNullOrEmpty(_options.Channel) ||
+                 !string.IsNullOrEmpty(_options.Commit) ||
+                 _options.DownloadSdk ||
+                 _options.DownloadRuntime ||
+                 _options.DownloadAspNet);
+
             if (_options.RootBuildId != 0)
             {
-                if (!string.IsNullOrEmpty(_options.RepoUri) ||
-                  !string.IsNullOrEmpty(_options.Channel) ||
-                  !string.IsNullOrEmpty(_options.Commit) ||
-                  _options.DownloadSdk ||
-                  _options.DownloadRuntime ||
-                  _options.DownloadAspNet)
+                if (hasNonIdRootBuildOptions || _options.RootBuildIds.Any())
                 {
                     Console.WriteLine("--id should not be specified with other options.");
                     return false;
                 }
                 return true;
             }
+            else if (_options.RootBuildIds.Any())
+            {
+                if (hasNonIdRootBuildOptions || _options.RootBuildId != 0)
+                {
+                    Console.WriteLine("--ids should not be specified with other options.");
+                    return false;
+                }
+                else if (_options.RootBuildIds.Any(id => id == 0))
```

Comment:
NIT: Why not validate that all are > 0?

Reply:
That's what this does. If any are 0, then it's an error.